src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -341,7 +315,13 @@ private Stream TryConnectToProcess(int nodeProcessId, int timeout, long hostHand
             // Try and connect to the process.
             string pipeName = "MSBuild" + nodeProcessId;
 
-            NamedPipeClientStream nodeStream = new NamedPipeClientStream(".", pipeName, PipeDirection.InOut, PipeOptions.Asynchronous);
+            PipeOptions pipeOptions = PipeOptions.Asynchronous;
+
+#if RUNTIME_TYPE_NETCORE
+            // TODO: Use enum value PipeOptions.CurrentUserOnly when NETStandard.Library is updated (Added in https://github.com/dotnet/corefx/issues/25427)
+            pipeOptions |= (PipeOptions)0x20000000; // PipeOptions.CurrentUserOnly;
```

Comment:
Just an FYI so everyone interested is aware: https://github.com/dotnet/corefx/issues/25427 is done, PR was merged last week.